--- trafshow-2.0/linux-include/machine/endian.h~	Sun Apr 13 14:22:58 1997
+++ trafshow-2.0/linux-include/machine/endian.h	Tue Dec 26 02:10:26 2000
@@ -1,4 +1,4 @@
-#include <bytesex.h>
+#include <endian.h>
 
 #ifndef NTOHL
 #if BYTE_ORDER == BIG_ENDIAN
--- trafshow-2.0/trafshow/show.c~	Tue Dec 26 02:11:49 2000
+++ trafshow-2.0/trafshow/show.c	Tue Dec 26 02:13:35 2000
@@ -22,10 +22,9 @@
 #include <netinet/in.h>
 #include <netinet/in_systm.h>
 #include <netinet/ip.h>
-#include <netinet/ip_var.h>
+#include <netinet/ip_icmp.h>
 #include <netinet/udp.h>
-#include <netinet/udp_var.h>
-#include <netinet/ip_tcp.h> 
+#include <netinet/tcp.h> 
 #include <netinet/if_ether.h>
 
 #include <pcap.h>
