diff -u -r trafshow-3.1/.trafshow trafshow-3.1-moj/.trafshow
--- trafshow-3.1/.trafshow	Mon Sep 22 06:58:48 1997
+++ trafshow-3.1-moj/.trafshow	Sun Sep 14 23:59:50 2003
@@ -20,12 +20,12 @@
 #domain1.foo.bar domain2.foo.bar any black:white
 #127.0.0.128/26 any any black:white
 
-#telnet	green
-#login	Green	# rlogin service
-#smtp	White
-#pop3	white
-#ftp	:Red
-#ftp-data White:Red
-#domain	Blue
-#nntp	yellow:White	# actually brown
-#*/icmp	black:white
+telnet	green
+login	Green	# rlogin service
+smtp	White
+pop3	white
+ftp	:Red
+ftp-data White:Red
+domain	Blue
+nntp	yellow:White	# actually brown
+*/icmp	black:white
diff -u -r trafshow-3.1/addrtoname.c trafshow-3.1-moj/addrtoname.c
--- trafshow-3.1/addrtoname.c	Mon Sep 15 00:00:51 2003
+++ trafshow-3.1/addrtoname.c	Sun Sep 14 23:12:25 2003
@@ -106,7 +106,7 @@
 	const int niflags = NI_NUMERICHOST;
 #endif
 
-	if (getnameinfo(sa, sa->sa_len, buf, sizeof(buf), NULL, 0,
+	if (getnameinfo(sa, sizeof (struct sockaddr), buf, sizeof(buf), NULL, 0,
 	    niflags) != 0)
 		strcpy(buf, "?");
 	return buf;
diff -u -r trafshow-3.1/display.c trafshow-3.1-moj/display.c
--- trafshow-3.1/display.c	Mon Sep 15 00:00:51 2003
+++ trafshow-3.1/display.c	Sun Sep 14 23:10:47 2003
@@ -487,13 +487,13 @@
 		sin = (struct sockaddr_in *)&te.src;
 		memset(sin, 0, sizeof(*sin));
 		sin->sin_family = AF_INET;
-		sin->sin_len = sizeof(*sin);
+//		sin->sin_len = sizeof(*sin);
 		sin->sin_addr.s_addr = ip->ip_src.s_addr;
 		sin->sin_port = sport;
 		sin = (struct sockaddr_in *)&te.dst;
 		memset(sin, 0, sizeof(*sin));
 		sin->sin_family = AF_INET;
-		sin->sin_len = sizeof(*sin);
+//		sin->sin_len = sizeof(*sin);
 		sin->sin_addr.s_addr = ip->ip_dst.s_addr;
 		sin->sin_port = dport;
 		te.proto = ip->ip_p;
@@ -541,13 +541,13 @@
 		sin6 = (struct sockaddr_in6 *)&te.src;
 		memset(sin6, 0, sizeof(*sin6));
 		sin6->sin6_family = AF_INET6;
-		sin6->sin6_len = sizeof(*sin6);
+//		sin6->sin6_len = sizeof(*sin6);
 		sin6->sin6_addr = ip6->ip6_src;
 		sin6->sin6_port = sport;
 		sin6 = (struct sockaddr_in6 *)&te.dst;
 		memset(sin6, 0, sizeof(*sin6));
 		sin6->sin6_family = AF_INET6;
-		sin6->sin6_len = sizeof(*sin6);
+//		sin6->sin6_len = sizeof(*sin6);
 		sin6->sin6_addr = ip6->ip6_dst;
 		sin6->sin6_port = dport;
 		te.proto = ip6->ip6_nxt;
@@ -557,12 +557,12 @@
 	else {	/* other than ip protocol packets */
 		sin = (struct sockaddr_in *)&te.src;
 		sin->sin_family = AF_INET;
-		sin->sin_len = sizeof(*sin);
+//		sin->sin_len = sizeof(*sin);
 		sin->sin_addr.s_addr = 0;
 		sin->sin_port = sport;
 		sin = (struct sockaddr_in *)&te.src;
 		sin->sin_family = AF_INET;
-		sin->sin_len = sizeof(*sin);
+//		sin->sin_len = sizeof(*sin);
 		sin->sin_addr.s_addr = 0;
 		sin->sin_port = dport;
 		te.proto = 0;
diff -u -r trafshow-3.1/getarptab.c trafshow-3.1-moj/getarptab.c
--- trafshow-3.1/getarptab.c	Mon Sep 15 00:00:51 2003
+++ trafshow-3.1/getarptab.c	Sun Sep 14 23:14:31 2003
@@ -20,7 +20,7 @@
 #include <sys/sysctl.h>
 
 #include <net/if.h>
-#include <net/if_dl.h>
+//#include <net/if_dl.h>
 #include <net/route.h>
 #include <netinet/in.h>
 #include <netinet/if_ether.h>
diff -u -r trafshow-3.1/interfaces.c trafshow-3.1/interfaces.c
--- trafshow-3.1/interfaces.c	Mon Sep 15 00:00:51 2003
+++ trafshow-3.1/interfaces.c	Sun Sep 14 23:16:46 2003
@@ -87,9 +87,6 @@
 		grab++;
 	switch (ntohs(((struct ether_header *)p)->ether_type)) {
 	case ETHERTYPE_IP:
-#ifdef INET6
-	case ETHERTYPE_IPV6:
-#endif
 		grab++;
 		break;
 	}
diff -u -r trafshow-3.1/linux/netinet/if_ether.h trafshow-3.1-moj/linux/netinet/if_ether.h
--- trafshow-3.1/linux/netinet/if_ether.h	Thu Sep 25 04:02:23 1997
+++ trafshow-3.1/linux/netinet/if_ether.h	Sun Sep 14 23:16:10 2003
@@ -53,6 +53,7 @@
 
 #define	ETHERTYPE_PUP		0x0200	/* PUP protocol */
 #define	ETHERTYPE_IP		0x0800	/* IP protocol */
+#define	ETHERTYPE_PPPOE		0x8864	/* IP protocol */
 #define ETHERTYPE_ARP		0x0806	/* Addr. resolution protocol */
 #define ETHERTYPE_REVARP	0x8035	/* reverse Addr. resolution protocol */
 
