--- trafshow-2.0/trafshow/show.c	Wed May 24 11:43:31 2000
+++ trafshow-2.0-new/trafshow/show.c	Wed May 24 11:32:14 2000
@@ -25,8 +25,7 @@
 #include <netinet/ip_var.h>
 #include <netinet/udp.h>
 #include <netinet/udp_var.h>
-#include <netinet/tcp.h>
-#include <netinet/tcpip.h>
+#include <netinet/ip_tcp.h> 
 #include <netinet/if_ether.h>
 
 #include <pcap.h>
--- trafshow-2.0/Makefile	Wed May 24 11:43:31 2000
+++ trafshow-2.0-new/Makefile	Wed May 24 11:23:56 2000
@@ -8,14 +8,14 @@
 FLAGS	= -O2
 endif
 DEFINE	=
-LIB	= ../lib/libbpft.a -L../libpcap -lncurses -ltermcap -lpcap
+LIB	= ../lib/libbpft.a -lpcap -lncurses -lpcap
 
-BINDIR	= /usr/local/bin
-MANDIR	= /usr/local/man/man1
+BINDIR	= /usr/bin
+MANDIR	= /usr/man/man1
 MODE	= 750
 GROUP	= wheel
 
-INCLUDE	= -I../include -I../linux-include -I../libpcap/linux-include -I../libpcap -I../libpcap/bpf -I/usr/include/bsd
+INCLUDE	= -I../include -I../linux-include 
 CFLAGS	= $(FLAGS) $(INCLUDE) $(DEFINE)
 DEPFLAGS= $(INCLUDE) $(DEFINE)
 ARFLAGS	= rc
